### DISCUSSION TITLE
Bootstrap add-ons features

### DISCUSSION THREAD

**Discussion by erickcomp** [unpoly/unpoly#567] [Q&A] [answered]
Hi! First of all, thanks for the amazing tool that you guys gave us.

So I'm starting to integrate unpoly into my app skeleton (which uses bootstrap 4). So I included the base unpoly files + bootstrap 4 ones. Right now, I'm using its compilers to activate some custom js when the elements are inserted into the DOM and it's working fine. But I noticed that my menus active classes vanished. I have a custom active menu logic because I have levels of sub-screens like this:

Admin > Users > Index (correctly active) URL: admin/users
Admin > Users > Edit User (wrongly removed active) URL: admin/users/edit/1

So, my question are:

1 - What does the bootstrap js + css gives us? (Docs)
2 - Is there a way for me to prevent the removal of my bootstrap active classes from my nav items?

Thanks!

**Comment by triskweline** [comment, marked as answer]
> 1 - What does the bootstrap js + css gives us? (Docs)

It configures Unpoly so it uses Bootstrap classes (instead of `up-` prefixed classes and attributes) where possible.

This is the code for Bootstrap 4:
https://github.com/unpoly/unpoly/blob/b79edeb81a292140986f0b552e98ec5935b4191e/src/unpoly-bootstrap4.sass
https://github.com/unpoly/unpoly/blob/b79edeb81a292140986f0b552e98ec5935b4191e/src/unpoly-bootstrap4.js#L3-L5

Note the highlighted lines in  the `.js`.

> 2 - Is there a way for me to prevent the removal of my bootstrap active classes from my nav items?

- If you remove `'active'` from `up.feedback.config.currentClasses` Unpoly will no longer set or remove this class. Unpoly will still set `.up-current` on links pointing to the current location.
- Unpoly only updates links set in a [navigational element](https://unpoly.com/up-nav). You could reconfigure `up.feedback.config.navSelectors` so it excludes the component where you manually control `.active`, e.g. by configuring `.nav:not(.my-custom-menu)`.
